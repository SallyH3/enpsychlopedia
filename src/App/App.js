import React, { Component } from 'react';
import '../Styles/Main.scss';
import Header from '../Header/Header.js';
import Genre from '../Genre/Genre.js';

class App extends Component {
  constructor() {
    super();
    this.state = {
      genres: [],
      bands: [],
      bandsById: []
    }
  }

   organizeData = () =>  {
    let data = this.state.genres.reduce((arr, genre) => {
        arr.push(this.state.bands.filter((band) => {
          return genre.id === band.genreId;
        }))
        return arr;
      }, []); 
      console.log(data)
    this.setState({ 
      bandsById: data
    })
  }

  componentDidMount() {
    fetch('http://whateverly-datasets.herokuapp.com/api/v1/psychMusic')
      .then(response => response.json())
      .then(music => this.setState({ genres: music.psychMusic }))
      .then(() => {this.organizeData()})
      .catch(err => console.log('music error', err))
    fetch('http://whateverly-datasets.herokuapp.com/api/v1/bands')
      .then(response => response.json())
      .then(band => this.setState({ bands: band.bands }))
      .then(() => {this.organizeData()})
      .catch(err => console.log('band error', err))
  }

  render() {
    return (
      <div className='app'>
        <Header />
        {this.state.genres.map((genre, index) => (
          <Genre key={index}
                 genre={genre}
                 sortedBands={this.state.bandsById[index]}
          />
        ))}
      </div>
    );
  }
}

export default App;
