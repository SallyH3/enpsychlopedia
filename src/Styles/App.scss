.app {
  background-color: lightblue;
  display: grid;
  grid-template-rows: 100px 340px 340px 340px;
}

