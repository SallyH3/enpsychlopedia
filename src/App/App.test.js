import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

  const genre = [{genre: "neoPsychedelia", origination: 1970, id: 1, characteristics: ["psychedelic pop", "jangly guitar rock", "heavily distorted", "free form jams", "recording experiments"]}, {genre: "stonerRock", origination: 1990, id: 2, characteristics: ["slow tempo", "bass heavy sound", "melodic vocals", "retro production", "groove laden"]}, {genre: "shoegazing", origination: 1980, id: 3, characteristics: ["guitar distortion", "delay effects", "obscured vocals", "ethereal layers", "reverb"]}]

  const bands = [
    {
      name: "Sparklehorse",
      origin: "Richmond Virginia",
      image: "https://i.imgur.com/ciMF4gb.jpg",
      members: [
        "Mark Linkous",
        "Scott Minor"
      ],
      genreId: 1,
      active: false,
      label: "Slow River",
      sideProjects: [
        "Danger Mouse",
        "Sol Seppy",
        "David Lynch",
        "Angela Faye Martin"
      ]
    },
    {
      name: "Electric Wizard",
      origin: "Dorset England",
      image: "https://i.imgur.com/oZCj9bF.jpg",
      members: [
        "Jus Oborn",
        "Liz Buckingham",
        "Clayton Burgess",
        "Simon Poole"
      ],
      genreId: 2,
      active: true,
      label: "Rise Above",
      sideProjects: [
        "Ramesses",
        "Sourvein",
        "Iron Monkey",
        "Teeth of Lions",
        "Rule the Divine",
        "Crippled Black Phoenix",
        "With the Dead"
      ]
    },
    {
      name: "Ride",
      origin: "Oxford England",
      image: "https://i.imgur.com/B41x1yH.jpg",
      members: [
        "Andy Bell",
        "Laurence Colbert",
        "Mark Gardener",
        "Steve Queralt"
      ],
      genreId: 3,
      active: true,
      label: "Creation",
      sideProjects: [
        "Oasis",
        "Beady Eye",
        "The Jesus and Mary Chain",
        "Gay Dad",
        "The Animalhouse",
        "Infinite Cassette"
      ]
      }  
  ]

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <App />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default states', () => {
    expect(wrapper.state()).toEqual({
      genres: [], 
      bands: [], 
      bandsById: []
    });
  });

  it('should update bandsById state when organizeData is invoked', () => {
    wrapper.setState({'bands': bands, 'genres': genre});
    expect(wrapper.state('bandsById')).toEqual([]);
    wrapper.instance().organizeData();
    expect(wrapper.state('bandsById')).toHaveLength(3);
  });
});
