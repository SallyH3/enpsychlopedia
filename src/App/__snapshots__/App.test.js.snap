// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="app"
>
  <Header />
</div>
`;

exports[`App should render to the DOM 1`] = `
<div
  className="app"
>
  <Header />
</div>
`;
